vignettes/delta-method.Rmd

```diff
@@ -119,3 +119,168 @@ Take the second derivative by subtracting each of the levels from the base level
 #### Continuous Variables
 
 Compute the second derivative of the link function and use that in place of the first derivative above. You need to explicitly compute the second derivative because you want an instantaneous rate of change as opposed to the rate of change over a range as with categorical variables above.
+
+# Example
+
+Say we're fitting a logistic regression using the `margex` data, and we're interested in the conditional treatment effect of treatment at various ages.
+
+```{r}
+library(modmarg)
+data(margex)
+lg <- glm(outcome ~ treatment * age, data = margex, family = 'binomial')
+summary(lg)
+```
+
+Where the `treatment` variable $\tau \in {0, 1}$. What is the average outcome when $\tau = 0$ or $\tau = 1$?
+
+Based on these results alone, we don't really know what the treatment effect is, both because we can't directly interpret the coefficients, because there's an interaction term, and because the effect of each covariate depends on the levels of the other covariates.
+
+## Point estimates
+
+The `predict` and `plogis` functions will get us partway there. Predict will calculate the linear predictors (in this case, setting $\text{treatment} = t$: 
+\[
+\hat{y}_i = \alpha + \beta_1 \cdot t + \beta_2 \text{age}_i + \beta_3 (t \cdot \text{age}_i)
+\]
+`plogis` is the inverse link function $f(z) = \frac{1}{1 + \exp(-z)}$, which transforms the linear predictors to the scale of the outcome variable (predicted probabilities). Finally, we can find the average predicted level.
+
+Combining these, we have:
+\[
+\hat{y}_{\tau = t} = \frac{1}{n} \sum_{i = 1}^n (1 + \exp(-\beta x_{i, \tau = t}))^{-1}
+\]
+We'll suppress the $\tau = t$ subscripts from here on.
+
+```{r}
+# Extract data from the model
+new <- lg$model
+# Set treatment to 0 for all observations
+new$treatment <- 0
+# Get predicted levels
+pred_ctl <- predict(lg, newdata = new)
+mean_pred_ctl <- mean(plogis(pred_ctl))
+
+# Once again, with treatment
+new <- lg$model
+# Set treatment to 1 for all observations
+new$treatment <- 1
+# 
+pred_treat <- predict(lg, newdata = new)
+mean_pred_treat <- mean(plogis(pred_treat))
+
+print(mean_pred_ctl)
+print(mean_pred_treat)
+```
+
+## Variance
+
+So far this is pretty straightforward, but we want to know the standard error of these estimates. Fortunately, our estimators $\hat{y}_{\tau = t}$ is a function of variables that obey the central limit theory - which means we can approximate their variance. Specifically, if the gradient $\nabla {\hat{y}_{\tau = t}}$ is defined as $\mathbf{J}$, then
```

Comment:
I think this is a little redundant with some of stuff above. Maybe we should combine so that explanation is in one place and example in other?